store/rootmulti/store.go

```diff
@@ -291,20 +291,25 @@ func (rs *Store) Commit() types.CommitID {
 	version := rs.lastCommitID.Version + 1
 	commitInfo := commitStores(version, rs.stores)
 
-	// Need to update atomically.
-	batch := rs.db.NewBatch()
-	defer batch.Close()
-	setCommitInfo(batch, version, commitInfo)
-	setLatestVersion(batch, version)
-	batch.Write()
-
-	// Prepare for next version.
-	commitID := types.CommitID{
-		Version: version,
-		Hash:    commitInfo.Hash(),
+	// update lastCommit only if this version was flushed to disk
+	if rs.pruningOpts.FlushVersion(version) {
```

Comment:
I'd do it the way around:

```golang
if !rs.pruningOpts.FlushVersion(version) {
    return rs.lastCommitID
}
```

So that you can:

- remove the `else` block
- reduce indentation of the statements currently in the `if` block